### PR TITLE
[rush] Check if package exists before publishing

### PR DESCRIPTION
**Pull request by qz2017**
The check existed in publish include-all path but not in change driven path. Now added.

### CODE REVIEW

File: apps/rush-lib/src/cli/actions/PublishAction.ts [hunk 1]
```diff
@@ -253,7 +253,7 @@ export class PublishAction extends BaseRushAction {
         for (const change of orderedChanges) {
           if (change.changeType && change.changeType > ChangeType.dependency) {
             const project: RushConfigurationProject | undefined = allPackages.get(change.packageName);
-            if (project) {
+            if (project && !this._packageExists(project)) {
```
Comment (pgonzal) [inline apps/rush-lib/src/cli/actions/PublishAction.ts]: >project && !this._packageExists(project)) [](start = 16, length = 41)

If the package already exists, we should probably print a big warning.  That is super weird and the end user needs to know about it.

Comment (pgonzal) [inline apps/rush-lib/src/cli/actions/PublishAction.ts]: >project [](start = 16, length = 7)

If every single package that we're trying to publish already exists, maybe that should be considered an error.

File: apps/rush-lib/src/cli/actions/PublishAction.ts [hunk 2]
```diff
@@ -326,6 +326,7 @@ export class PublishAction extends BaseRushAction {
         const registryUrl: string = this._registryUrl.value;
         env['npm_config_registry'] = registryUrl; // tslint:disable-line:no-string-literal
         registry = registryUrl.substring(registryUrl.indexOf('//'));
+        args.push(`--registry ${registryUrl}`);
```
Comment (pgonzal) [inline apps/rush-lib/src/cli/actions/PublishAction.ts]: >${registryUrl} [](start = 30, length = 14)

This is a separate arg, so normally it should be pushed separately.  Perhaps execCommand() joins everything with spaces and lets the shell reparse it, but that may not always be the case

File: apps/rush-lib/src/cli/actions/PublishAction.ts [hunk 3]
```diff
@@ -254,7 +254,13 @@ export class PublishAction extends BaseRushAction {
           if (change.changeType && change.changeType > ChangeType.dependency) {
             const project: RushConfigurationProject | undefined = allPackages.get(change.packageName);
             if (project) {
-              this._npmPublish(change.packageName, project.projectFolder);
+              if (!this._packageExists(project)) {
+                this._npmPublish(change.packageName, project.projectFolder);
+              } else {
+                console.log(`Skip ${change.packageName}. Package exists.`);
```
Comment (cliffkoh) [inline apps/rush-lib/src/cli/actions/PublishAction.ts]: nit: Skipping instead of Skip

**Comment by pgonzal** [comment]
@qz2017 this PR has been open for a long time -- do you still have plans to merge it?

**Pull request merged**